docs/LabHome/Onboarding/coding-guide.md

```diff
@@ -0,0 +1,91 @@
+  * Do not mix code with your data; instead, make separate folders for code and data.
+  * Avoid using GitHub to track image data or any data with large file size, or any data you will frequently change. Use `DataLad` to track your data.
+  * Keep raw data in a separate folder. Save data derivatives in another folder.
+* Tools: VS Code provides a great UI for coding (e.g., Python, bash), and it is also very helpful for git version control; R Studio provides a great UI for coding in R. Tips: use the "outline" in those tools to jump across sections in scripts.
+
+
+## Coding itself
+
+### Just before coding
+Think about the function of the code you want to achieve. What language will be efficient to achieve your goal here? You can have more than one language of code in one project! Python might be easy for image processing; R would be easy for data wrangling; bash scripts are easy for submitting jobs on clusters.
+
+If the code involves a few steps, you might get lost when coding/focusing on one step. Consider writing down the list of the steps as comments before you start coding it.
+
+### Reduce duplicated code
+If you have duplicated code, it might be a good time to translate them into a function, so that in the future, you only need to edit the function once. Otherwise, if you want to edit one of the versions, you will also have to update the other one. In other words, it's easier for version control.
+
+### Reduce duplicated list or table as input
+Sometimes you hope to define a list of brain regions in your analysis, and you'll use it repeatedly in other scripts too. Define this list in a text file and save it in the Git repository if the file size is not big. For table (e.g., full names, abbreviations, color schemes, etc of brain regions), you can define it in a CSV file, and also save it in the Git repository if that's not big. When you want to use this, just `source` the text file or directly load the CSV file.
+
+
+### Data wrangling
+The data analysis part after (pre)processing often involve data wrangling. We **highly** recommend using R to handle the data wrangling part. [Tidy data](https://r4ds.had.co.nz/tidy-data.html) is often very helpful here. It is important to note that, although some columns seem redundant in the tidy data, it is very important to include them so that future data wrangling will be easy, and it'll be easy for sanity checks too (i.e., you'll know the exact information of the observation, e.g., from which subject, which session, which method, etc)
+
+### Reduce manual work and additional brain work!!!
+This is very important. Manual work e.g., manually counting or changing numbers increases risk of mistakes and reduces the confidence of what you reported in the manuscript. In addition, it will also increase the burden for the replication buddy - they need to more carefully check it to make sure what you reported are consistent with the original output, especially if they are in different format.
```

Comment:
```suggestion
This is very important.
Manual work (e.g., manually counting or changing numbers) increases risk of mistakes and reduces the confidence of what you reported in the manuscript.
In addition, it will also increase the burden for the reproducibility buddy - they need to more carefully check it to make sure what you reported are consistent with the original output, especially if they are in a different format.
```